src/components/test/common-tests.js

```diff
@@ -237,3 +238,40 @@ export function testSimpleComponent(Component) {
     });
   });
 }
+
+/**
+ * Common tests for simple design components
+ *
+ * @param {TransitionComponent} Component
+ * @param {CommonTestOpts} opts
+ */
+export function testTransitionComponent(
+  Component,
+  { componentName, createContent = createComponent } = {}
+) {
+  const displayName = componentName ?? Component.displayName ?? Component.name;
+
+  describe(`Common transition component functionality for ${displayName}`, () => {
+    it('renders', () => {
+      const wrapper = createComponent(Component);
+      assert.isTrue(wrapper.find(Component).exists());
+    });
+
+    ['in', 'out'].forEach(direction => {
+      it('should handle unmounting on either direction', () => {
```

Comment:
Can you help me understand exactly what this is testing (specifically, why unmounting is tested)?

Reply:
See https://github.com/hypothesis/frontend-shared/pull/1002/files/f1eabd4ac120780602a06b28347481136ece4d0e#r1179255457